### DISCUSSION TITLE
kubernetes or knative issue?

### DISCUSSION THREAD

**Discussion by stormwalkerec** [vitobotta/hetzner-k3s#212] [Q&A] [answered]
Greetings. Total noob here but learning.

Stock install of k3s and knative.

using patch:
`kubectl patch service kourier \
  --namespace kourier-system \
  --type merge \
  --patch '{"metadata":{"annotations":{"load-balancer.hetzner.cloud/location":"fsn1","load-balancer.hetzner.cloud/use-private-ip":"true","load-balancer.hetzner.cloud/name":"knative-lb","load-balancer.hetzner.cloud/hostname":"knative.domain.com","load-balancer.hetzner.cloud/uses-proxyprotocol":"true"}}}'`

The load balancer created by kourier adds two lb services in hetznet lb:
tcp:80 -> 30532 (port changes with each install)
tcp:443 -> 32430  (port also changes with each install)

tcp:80 -> 30532 - one is all green
tcp:443 -> 32430 -  is all red.

Quick firewall edit to debug.
Doing a port open check on the worker node instances the 30532 shows as open, and 32430 shows as closed.

Knative install is essentially install serving crds, serving core, kourier & patch to get working. nothing about setting up potential ssl certs

So trying to figure out if the http/https ports for each instances is created by kubernetes and informs the lb when knative/koutier does its thing, or is it knative that tells the instance to open that port (or installs a gateway pod with the settings). I assume potentially that there may need to be some certificate installed/shared to all causing the https port not to be open/failed config... that's the avenue I'm exploring right now, but any help would be appreciated. But could be a trap for noob users, hence the ask.

Thanks !!

**Comment by stormwalkerec** [comment]
OK, trying out the Istio option with a modified patch,

tcp:80 - all red
tcp:443 - all red
tcp15021 - all green

if it were a cert issue, then the http port would also have been green.

I have found that istio installs an ingress gateway on instances that receives on those ports, not sure now if its a lb settings issue, or knative/kubernetes at this point.... continuing

**Comment by vitobotta** [comment]
I can't help yet because I haven't touched knative in general in many months. I will have to try it.

**Reply by stormwalkerec** [reply]
thanks, no rush.... I learn fastest when trying to find the issue... mainly posting in the event its one of those typical new user cases that may be a quick setting ...

**Comment by stormwalkerec** [comment, marked as answer]
Its a knative settings issue. [See the guide to install knative](https://github.com/vitobotta/hetzner-k3s/discussions/215)
